src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -221,56 +225,89 @@ object DeployPlugin extends AutoPlugin {
+          )
+
+          // Shell-friendly version of rsync command, with rsh value quoted.
+          val quotedRsync = rsyncCommand.patch(
+            2, Seq("--rsh=" + sshCommand.mkString("\"", " ", "\"")), 1
+          ).mkString(" ")
+          log.info("Running " + quotedRsync + " . . .")
+          if (Process(rsyncCommand).! != 0) {
+            throw new IllegalArgumentException(s"Error running rsync to host '$deployHost'.")
+          }
+
+          // Now, ssh to the remote host and run the restart script.
+          val restartScript = deployDirectory + "/" + hostConfig.startupScript
+          val restartArgs = if (hostConfig.startupArgs.nonEmpty) {
+            "--" +: hostConfig.startupArgs
+          } else {
+            Seq()
+          }
+
+          val restartCommand = Seq.concat(
+            sshCommand,
+            Seq(deployHost, restartScript, "restart"),
+            restartArgs
+          )
+          log.info("Running " + restartCommand.mkString(" ") + " . . .")
+          if (Process(restartCommand).! != 0) {
+            throw new IllegalArgumentException(s"Error running restart command on host '$deployHost'.")
+          }
+        }
+      }
     }
-    val deployHost = configMap("deploy.host")
-    val deployDirectory = configMap("deploy.directory")
-
-    val rsyncDirs = deployDirs.value map (name => s"--include=/${name}")
-    val rsyncCommand = Seq("rsync", "-vcrtzP", "--rsh=" + sshCommand.mkString(" ")) ++ rsyncDirs ++
-      Seq(
-        "--exclude=/*",
-        "--delete",
-        universalStagingDir.getPath + "/",
-        deployHost + ":" + deployDirectory
-      )
-
-    // Shell-friendly version of rsync command, with rsh value quoted.
-    val quotedRsync = rsyncCommand.patch(
-      2, Seq("--rsh=" + sshCommand.mkString("\"", " ", "\"")), 1
-    ).mkString(" ")
-    log.info("Running " + quotedRsync + " . . .")
-    if (Process(rsyncCommand).! != 0) {
-      throw new IllegalArgumentException("Error running rsync.")
+
+    // Accumulate all error messages thrown when trying to deploy.
+    val accumulatedErrors = Future.fold[Try[Unit], Seq[Throwable]](deploys)(Seq.empty) {
+      case (acc, Success(_)) => acc
+      case (acc, Failure(e)) => acc :+ e
     }
 
-    // Now, ssh to the remote host and run the restart script.
-    val restartScript = deployDirectory + "/" + configMap("deploy.startup_script")
-    val restartCommand = sshCommand ++ Seq(deployHost, restartScript, "restart")
-    log.info("Running " + restartCommand.mkString(" ") + " . . .")
-    if (Process(restartCommand).! != 0) {
-      throw new IllegalArgumentException("Error running restart command.")
+    // Throw an error if any deploy commands failed.
+    val deployStatus = accumulatedErrors map { errorSeq =>
+      if (errorSeq.nonEmpty) {
+        throw new IllegalArgumentException(
+          "Hit the following errors during deployment:" +
+            (errorSeq map { _.getMessage } mkString ("\n", "\n", "\n"))
```

Comment:
I personally find the three-arg `mkString` a little awkward.

I'd probably do:
```scala
val errors = errorSeq.mkString("\n")
throw new Exception(s"Hit the following errors during deployment:\n$errors\n")
```
I'd also probably format more like `(err, err, err)` instead of putting newlines in - they are a bit hard-to-read, and don't make it obvious these came from a list.